scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/AzureBlobDownloader.java

```diff
@@ -0,0 +1,222 @@
+            createDirIfNotExists(file);
+            downloadContainerBlobs(containerURL);
+        }
+
+        return (Serializable) Arrays.asList(outputPath);
+    }
+
+    private void downloadContainerBlobs(ContainerURL containerURL) {
+        // Each ContainerURL.listBlobsFlatSegment call return up to maxResults (maxResults=10 passed into ListBlobOptions below).
+        // To list all Blobs, we are creating a helper static method called downloadAllBlobs,
+        // and calling it after the initial listBlobsFlatSegment call
+        ListBlobsOptions options = new ListBlobsOptions(null, null, 1);
+
+        containerURL.listBlobsFlatSegment(null, options)
+                    .flatMap(containerListBlobFlatSegmentResponse -> downloadAllBlobs(containerURL,
+                                                                                      containerListBlobFlatSegmentResponse))
+                    .blockingGet();
+        getOut().println("Downloading of the container blobs completed.");
+    }
+
+    private Single<ContainerListBlobFlatSegmentResponse> downloadAllBlobs(ContainerURL url,
+            ContainerListBlobFlatSegmentResponse response) {
+        // Process the blobs returned in this result segment (if the segment is empty, blobs() will be null.
+        if (response.body().segment() != null) {
+            for (BlobItem b : response.body().segment().blobItems()) {
+                File azureBlobFile = new File(Paths.get(outputPath, b.name()).toString());
+                if (!azureBlobFile.getParentFile().exists()) {
+                    azureBlobFile.getParentFile().mkdirs();
+                }
+                downloadBlob(azureBlobFile, b.name());
+            }
+        } else {
+            getOut().println("There are no more blobs to list off.");
+        }
+
+        // If there is not another segment, return this response as the final response.
+        if (response.body().nextMarker() == null) {
+            return Single.just(response);
+        } else {
+            /*
+             * IMPORTANT: ListBlobsFlatSegment returns the start of the next segment; you MUST use
+             * this to get the next
+             * segment (after processing the current result segment
+             */
+
+            String nextMarker = response.body().nextMarker();
+
+            /*
+             * The presence of the marker indicates that there are more blobs to list, so we make
+             * another call to
+             * listBlobsFlatSegment and pass the result through this helper function.
+             */
+
+            return url.listBlobsFlatSegment(nextMarker, new ListBlobsOptions(null, null, 1))
+                      .flatMap(containersListBlobFlatSegmentResponse -> downloadAllBlobs(url,
+                                                                                         containersListBlobFlatSegmentResponse));
+        }
+    }
+
+    private void downloadBlob(File destinationFile, String blobName) {
```

Comment:
It is confusing to have `blobName` as an method argument here because is also the class attribute. Rename the parameter or remove it please.